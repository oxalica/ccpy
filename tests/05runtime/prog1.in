def __builtin__global_get(name):
  g = __intrinsic__get_global0()
  return __intrinsic__dict_get3(g, name, None)

def __builtin__call(f, args):
  return __intrinsic__v_call2(f, args)

def __builtin__add(a, b):
  return __intrinsic__str_concat2(a, b)

def __builtin__index(obj, idx):
  return __intrinsic__tuple_idx2(obj, idx)


def print_s(s=''):
  s = s + '\n'
  __intrinsic__print1(s)
  return 233

def print(*xs):
  print_s(repr(xs))

def repr(obj):
  return __intrinsic__repr1(obj)

def f():
  def g(a=()):
    __intrinsic__tuple_splice4(a, 0, 0, (1,))
    return a
  return g

def g(a="a", *b): print(a, b)

ret = print_s('Hello, world!')
print_s('ret = ' + repr(ret))
print_s()

a = b = 1
print(a, b)
b = a, b
print(a, b)
a, b = b, a
print(a, b)
print_s()

g()
g('Hello')
g('Hello', 'world')
g('Hello', 'cruel', 'world')
print_s()

a = f()
b = f()
print(a(), a())
print(b())
print(a())

def __builtin__not(obj):
  print_s('__builtin__not')
  return __intrinsic__not1(obj)

def cond(a):
  s = ''
  print_s('--')
  if __intrinsic__not1(a): # No builtin call
    s = s + 'F'
  else:
    s = s + 'T'
  print_s('--')
  if not a: # 2 builtin call
    s = s + 'F'
  else:
    s = s + 'T'
  print_s('--')
  s = s + ('T' if a else 'F') # 1 builtin call
  return s

print_s()
print(cond(True, True), cond(False, True)) # 4 builtin call
print(cond(1), cond(0)) # `0` as Truth due to `__builtin__not` here

