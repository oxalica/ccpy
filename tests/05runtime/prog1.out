Start(Local size = 7):
0: %0 = __intrinsic__tuple_make_()
1: %0 = Closure #1() defaults = %0
2: %1 = __intrinsic__get_global0()
3: %2 = "__builtin__global_get"
4: %0 = __intrinsic__dict_set3(%1, %2, %0)
5: %0 = __intrinsic__tuple_make_()
6: %0 = Closure #2() defaults = %0
7: %1 = __intrinsic__get_global0()
8: %2 = "__builtin__call"
9: %0 = __intrinsic__dict_set3(%1, %2, %0)
10: %0 = __intrinsic__tuple_make_()
11: %0 = Closure #3() defaults = %0
12: %1 = __intrinsic__get_global0()
13: %2 = "__builtin__add"
14: %0 = __intrinsic__dict_set3(%1, %2, %0)
15: %0 = __intrinsic__tuple_make_()
16: %0 = Closure #4() defaults = %0
17: %1 = __intrinsic__get_global0()
18: %2 = "__builtin__index"
19: %0 = __intrinsic__dict_set3(%1, %2, %0)
20: %0 = ""
21: %0 = __intrinsic__tuple_make_(%0)
22: %0 = Closure #5() defaults = %0
23: %1 = __intrinsic__get_global0()
24: %2 = "print_s"
25: %0 = __intrinsic__dict_set3(%1, %2, %0)
26: %0 = __intrinsic__tuple_make_()
27: %0 = Closure #6() defaults = %0
28: %1 = __intrinsic__get_global0()
29: %2 = "print"
30: %0 = __intrinsic__dict_set3(%1, %2, %0)
31: %0 = __intrinsic__tuple_make_()
32: %0 = Closure #7() defaults = %0
33: %1 = __intrinsic__get_global0()
34: %2 = "repr"
35: %0 = __intrinsic__dict_set3(%1, %2, %0)
36: %0 = __intrinsic__tuple_make_()
37: %0 = Closure #8() defaults = %0
38: %1 = __intrinsic__get_global0()
39: %2 = "f"
40: %0 = __intrinsic__dict_set3(%1, %2, %0)
41: %0 = "a"
42: %0 = __intrinsic__tuple_make_(%0)
43: %0 = Closure #10() defaults = %0
44: %1 = __intrinsic__get_global0()
45: %2 = "g"
46: %0 = __intrinsic__dict_set3(%1, %2, %0)
47: %0 = "print_s"
48: %1 = __intrinsic__get_global0()
49: %2 = "__builtin__global_get"
50: %3 = None
51: %1 = __intrinsic__dict_get3(%1, %2, %3)
52: %0 = __intrinsic__tuple_make_(%0)
53: %0 = __intrinsic__v_call2(%1, %0)
54: %1 = "Hello, world!"
55: %1 = __intrinsic__tuple_make_(%1)
56: %2 = __intrinsic__get_global0()
57: %3 = "__builtin__call"
58: %4 = None
59: %2 = __intrinsic__dict_get3(%2, %3, %4)
60: %0 = __intrinsic__tuple_make_(%0, %1)
61: %0 = __intrinsic__v_call2(%2, %0)
62: %1 = __intrinsic__get_global0()
63: %2 = "ret"
64: %1 = __intrinsic__dict_set3(%1, %2, %0)
65: %0 = "print_s"
66: %1 = __intrinsic__get_global0()
67: %2 = "__builtin__global_get"
68: %3 = None
69: %1 = __intrinsic__dict_get3(%1, %2, %3)
70: %0 = __intrinsic__tuple_make_(%0)
71: %0 = __intrinsic__v_call2(%1, %0)
72: %1 = "ret = "
73: %2 = "repr"
74: %3 = __intrinsic__get_global0()
75: %4 = "__builtin__global_get"
76: %5 = None
77: %3 = __intrinsic__dict_get3(%3, %4, %5)
78: %2 = __intrinsic__tuple_make_(%2)
79: %2 = __intrinsic__v_call2(%3, %2)
80: %3 = "ret"
81: %4 = __intrinsic__get_global0()
82: %5 = "__builtin__global_get"
83: %6 = None
84: %4 = __intrinsic__dict_get3(%4, %5, %6)
85: %3 = __intrinsic__tuple_make_(%3)
86: %3 = __intrinsic__v_call2(%4, %3)
87: %3 = __intrinsic__tuple_make_(%3)
88: %4 = __intrinsic__get_global0()
89: %5 = "__builtin__call"
90: %6 = None
91: %4 = __intrinsic__dict_get3(%4, %5, %6)
92: %2 = __intrinsic__tuple_make_(%2, %3)
93: %2 = __intrinsic__v_call2(%4, %2)
94: %3 = __intrinsic__get_global0()
95: %4 = "__builtin__add"
96: %5 = None
97: %3 = __intrinsic__dict_get3(%3, %4, %5)
98: %1 = __intrinsic__tuple_make_(%1, %2)
99: %1 = __intrinsic__v_call2(%3, %1)
100: %1 = __intrinsic__tuple_make_(%1)
101: %2 = __intrinsic__get_global0()
102: %3 = "__builtin__call"
103: %4 = None
104: %2 = __intrinsic__dict_get3(%2, %3, %4)
105: %0 = __intrinsic__tuple_make_(%0, %1)
106: %0 = __intrinsic__v_call2(%2, %0)
107: %0 = "print_s"
108: %1 = __intrinsic__get_global0()
109: %2 = "__builtin__global_get"
110: %3 = None
111: %1 = __intrinsic__dict_get3(%1, %2, %3)
112: %0 = __intrinsic__tuple_make_(%0)
113: %0 = __intrinsic__v_call2(%1, %0)
114: %1 = __intrinsic__tuple_make_()
115: %2 = __intrinsic__get_global0()
116: %3 = "__builtin__call"
117: %4 = None
118: %2 = __intrinsic__dict_get3(%2, %3, %4)
119: %0 = __intrinsic__tuple_make_(%0, %1)
120: %0 = __intrinsic__v_call2(%2, %0)
121: %0 = 1
122: %1 = __intrinsic__get_global0()
123: %2 = "a"
124: %1 = __intrinsic__dict_set3(%1, %2, %0)
125: %1 = __intrinsic__get_global0()
126: %2 = "b"
127: %1 = __intrinsic__dict_set3(%1, %2, %0)
128: %0 = "print"
129: %1 = __intrinsic__get_global0()
130: %2 = "__builtin__global_get"
131: %3 = None
132: %1 = __intrinsic__dict_get3(%1, %2, %3)
133: %0 = __intrinsic__tuple_make_(%0)
134: %0 = __intrinsic__v_call2(%1, %0)
135: %1 = "a"
136: %2 = __intrinsic__get_global0()
137: %3 = "__builtin__global_get"
138: %4 = None
139: %2 = __intrinsic__dict_get3(%2, %3, %4)
140: %1 = __intrinsic__tuple_make_(%1)
141: %1 = __intrinsic__v_call2(%2, %1)
142: %2 = "b"
143: %3 = __intrinsic__get_global0()
144: %4 = "__builtin__global_get"
145: %5 = None
146: %3 = __intrinsic__dict_get3(%3, %4, %5)
147: %2 = __intrinsic__tuple_make_(%2)
148: %2 = __intrinsic__v_call2(%3, %2)
149: %1 = __intrinsic__tuple_make_(%1, %2)
150: %2 = __intrinsic__get_global0()
151: %3 = "__builtin__call"
152: %4 = None
153: %2 = __intrinsic__dict_get3(%2, %3, %4)
154: %0 = __intrinsic__tuple_make_(%0, %1)
155: %0 = __intrinsic__v_call2(%2, %0)
156: %0 = "a"
157: %1 = __intrinsic__get_global0()
158: %2 = "__builtin__global_get"
159: %3 = None
160: %1 = __intrinsic__dict_get3(%1, %2, %3)
161: %0 = __intrinsic__tuple_make_(%0)
162: %0 = __intrinsic__v_call2(%1, %0)
163: %1 = "b"
164: %2 = __intrinsic__get_global0()
165: %3 = "__builtin__global_get"
166: %4 = None
167: %2 = __intrinsic__dict_get3(%2, %3, %4)
168: %1 = __intrinsic__tuple_make_(%1)
169: %1 = __intrinsic__v_call2(%2, %1)
170: %0 = __intrinsic__tuple_make_(%0, %1)
171: %1 = __intrinsic__get_global0()
172: %2 = "b"
173: %1 = __intrinsic__dict_set3(%1, %2, %0)
174: %0 = "print"
175: %1 = __intrinsic__get_global0()
176: %2 = "__builtin__global_get"
177: %3 = None
178: %1 = __intrinsic__dict_get3(%1, %2, %3)
179: %0 = __intrinsic__tuple_make_(%0)
180: %0 = __intrinsic__v_call2(%1, %0)
181: %1 = "a"
182: %2 = __intrinsic__get_global0()
183: %3 = "__builtin__global_get"
184: %4 = None
185: %2 = __intrinsic__dict_get3(%2, %3, %4)
186: %1 = __intrinsic__tuple_make_(%1)
187: %1 = __intrinsic__v_call2(%2, %1)
188: %2 = "b"
189: %3 = __intrinsic__get_global0()
190: %4 = "__builtin__global_get"
191: %5 = None
192: %3 = __intrinsic__dict_get3(%3, %4, %5)
193: %2 = __intrinsic__tuple_make_(%2)
194: %2 = __intrinsic__v_call2(%3, %2)
195: %1 = __intrinsic__tuple_make_(%1, %2)
196: %2 = __intrinsic__get_global0()
197: %3 = "__builtin__call"
198: %4 = None
199: %2 = __intrinsic__dict_get3(%2, %3, %4)
200: %0 = __intrinsic__tuple_make_(%0, %1)
201: %0 = __intrinsic__v_call2(%2, %0)
202: %0 = "b"
203: %1 = __intrinsic__get_global0()
204: %2 = "__builtin__global_get"
205: %3 = None
206: %1 = __intrinsic__dict_get3(%1, %2, %3)
207: %0 = __intrinsic__tuple_make_(%0)
208: %0 = __intrinsic__v_call2(%1, %0)
209: %1 = "a"
210: %2 = __intrinsic__get_global0()
211: %3 = "__builtin__global_get"
212: %4 = None
213: %2 = __intrinsic__dict_get3(%2, %3, %4)
214: %1 = __intrinsic__tuple_make_(%1)
215: %1 = __intrinsic__v_call2(%2, %1)
216: %0 = __intrinsic__tuple_make_(%0, %1)
217: %1 = 0
218: %2 = __intrinsic__get_global0()
219: %3 = "__builtin__index"
220: %4 = None
221: %2 = __intrinsic__dict_get3(%2, %3, %4)
222: %1 = __intrinsic__tuple_make_(%0, %1)
223: %1 = __intrinsic__v_call2(%2, %1)
224: %2 = __intrinsic__get_global0()
225: %3 = "a"
226: %1 = __intrinsic__dict_set3(%2, %3, %1)
227: %1 = 1
228: %2 = __intrinsic__get_global0()
229: %3 = "__builtin__index"
230: %4 = None
231: %2 = __intrinsic__dict_get3(%2, %3, %4)
232: %1 = __intrinsic__tuple_make_(%0, %1)
233: %1 = __intrinsic__v_call2(%2, %1)
234: %2 = __intrinsic__get_global0()
235: %3 = "b"
236: %1 = __intrinsic__dict_set3(%2, %3, %1)
237: %0 = "print"
238: %1 = __intrinsic__get_global0()
239: %2 = "__builtin__global_get"
240: %3 = None
241: %1 = __intrinsic__dict_get3(%1, %2, %3)
242: %0 = __intrinsic__tuple_make_(%0)
243: %0 = __intrinsic__v_call2(%1, %0)
244: %1 = "a"
245: %2 = __intrinsic__get_global0()
246: %3 = "__builtin__global_get"
247: %4 = None
248: %2 = __intrinsic__dict_get3(%2, %3, %4)
249: %1 = __intrinsic__tuple_make_(%1)
250: %1 = __intrinsic__v_call2(%2, %1)
251: %2 = "b"
252: %3 = __intrinsic__get_global0()
253: %4 = "__builtin__global_get"
254: %5 = None
255: %3 = __intrinsic__dict_get3(%3, %4, %5)
256: %2 = __intrinsic__tuple_make_(%2)
257: %2 = __intrinsic__v_call2(%3, %2)
258: %1 = __intrinsic__tuple_make_(%1, %2)
259: %2 = __intrinsic__get_global0()
260: %3 = "__builtin__call"
261: %4 = None
262: %2 = __intrinsic__dict_get3(%2, %3, %4)
263: %0 = __intrinsic__tuple_make_(%0, %1)
264: %0 = __intrinsic__v_call2(%2, %0)
265: %0 = "print_s"
266: %1 = __intrinsic__get_global0()
267: %2 = "__builtin__global_get"
268: %3 = None
269: %1 = __intrinsic__dict_get3(%1, %2, %3)
270: %0 = __intrinsic__tuple_make_(%0)
271: %0 = __intrinsic__v_call2(%1, %0)
272: %1 = __intrinsic__tuple_make_()
273: %2 = __intrinsic__get_global0()
274: %3 = "__builtin__call"
275: %4 = None
276: %2 = __intrinsic__dict_get3(%2, %3, %4)
277: %0 = __intrinsic__tuple_make_(%0, %1)
278: %0 = __intrinsic__v_call2(%2, %0)
279: %0 = "g"
280: %1 = __intrinsic__get_global0()
281: %2 = "__builtin__global_get"
282: %3 = None
283: %1 = __intrinsic__dict_get3(%1, %2, %3)
284: %0 = __intrinsic__tuple_make_(%0)
285: %0 = __intrinsic__v_call2(%1, %0)
286: %1 = __intrinsic__tuple_make_()
287: %2 = __intrinsic__get_global0()
288: %3 = "__builtin__call"
289: %4 = None
290: %2 = __intrinsic__dict_get3(%2, %3, %4)
291: %0 = __intrinsic__tuple_make_(%0, %1)
292: %0 = __intrinsic__v_call2(%2, %0)
293: %0 = "g"
294: %1 = __intrinsic__get_global0()
295: %2 = "__builtin__global_get"
296: %3 = None
297: %1 = __intrinsic__dict_get3(%1, %2, %3)
298: %0 = __intrinsic__tuple_make_(%0)
299: %0 = __intrinsic__v_call2(%1, %0)
300: %1 = "Hello"
301: %1 = __intrinsic__tuple_make_(%1)
302: %2 = __intrinsic__get_global0()
303: %3 = "__builtin__call"
304: %4 = None
305: %2 = __intrinsic__dict_get3(%2, %3, %4)
306: %0 = __intrinsic__tuple_make_(%0, %1)
307: %0 = __intrinsic__v_call2(%2, %0)
308: %0 = "g"
309: %1 = __intrinsic__get_global0()
310: %2 = "__builtin__global_get"
311: %3 = None
312: %1 = __intrinsic__dict_get3(%1, %2, %3)
313: %0 = __intrinsic__tuple_make_(%0)
314: %0 = __intrinsic__v_call2(%1, %0)
315: %1 = "Hello"
316: %2 = "world"
317: %1 = __intrinsic__tuple_make_(%1, %2)
318: %2 = __intrinsic__get_global0()
319: %3 = "__builtin__call"
320: %4 = None
321: %2 = __intrinsic__dict_get3(%2, %3, %4)
322: %0 = __intrinsic__tuple_make_(%0, %1)
323: %0 = __intrinsic__v_call2(%2, %0)
324: %0 = "g"
325: %1 = __intrinsic__get_global0()
326: %2 = "__builtin__global_get"
327: %3 = None
328: %1 = __intrinsic__dict_get3(%1, %2, %3)
329: %0 = __intrinsic__tuple_make_(%0)
330: %0 = __intrinsic__v_call2(%1, %0)
331: %1 = "Hello"
332: %2 = "cruel"
333: %3 = "world"
334: %1 = __intrinsic__tuple_make_(%1, %2, %3)
335: %2 = __intrinsic__get_global0()
336: %3 = "__builtin__call"
337: %4 = None
338: %2 = __intrinsic__dict_get3(%2, %3, %4)
339: %0 = __intrinsic__tuple_make_(%0, %1)
340: %0 = __intrinsic__v_call2(%2, %0)
341: %0 = "print_s"
342: %1 = __intrinsic__get_global0()
343: %2 = "__builtin__global_get"
344: %3 = None
345: %1 = __intrinsic__dict_get3(%1, %2, %3)
346: %0 = __intrinsic__tuple_make_(%0)
347: %0 = __intrinsic__v_call2(%1, %0)
348: %1 = __intrinsic__tuple_make_()
349: %2 = __intrinsic__get_global0()
350: %3 = "__builtin__call"
351: %4 = None
352: %2 = __intrinsic__dict_get3(%2, %3, %4)
353: %0 = __intrinsic__tuple_make_(%0, %1)
354: %0 = __intrinsic__v_call2(%2, %0)
355: %0 = "f"
356: %1 = __intrinsic__get_global0()
357: %2 = "__builtin__global_get"
358: %3 = None
359: %1 = __intrinsic__dict_get3(%1, %2, %3)
360: %0 = __intrinsic__tuple_make_(%0)
361: %0 = __intrinsic__v_call2(%1, %0)
362: %1 = __intrinsic__tuple_make_()
363: %2 = __intrinsic__get_global0()
364: %3 = "__builtin__call"
365: %4 = None
366: %2 = __intrinsic__dict_get3(%2, %3, %4)
367: %0 = __intrinsic__tuple_make_(%0, %1)
368: %0 = __intrinsic__v_call2(%2, %0)
369: %1 = __intrinsic__get_global0()
370: %2 = "a"
371: %1 = __intrinsic__dict_set3(%1, %2, %0)
372: %0 = "f"
373: %1 = __intrinsic__get_global0()
374: %2 = "__builtin__global_get"
375: %3 = None
376: %1 = __intrinsic__dict_get3(%1, %2, %3)
377: %0 = __intrinsic__tuple_make_(%0)
378: %0 = __intrinsic__v_call2(%1, %0)
379: %1 = __intrinsic__tuple_make_()
380: %2 = __intrinsic__get_global0()
381: %3 = "__builtin__call"
382: %4 = None
383: %2 = __intrinsic__dict_get3(%2, %3, %4)
384: %0 = __intrinsic__tuple_make_(%0, %1)
385: %0 = __intrinsic__v_call2(%2, %0)
386: %1 = __intrinsic__get_global0()
387: %2 = "b"
388: %1 = __intrinsic__dict_set3(%1, %2, %0)
389: %0 = "print"
390: %1 = __intrinsic__get_global0()
391: %2 = "__builtin__global_get"
392: %3 = None
393: %1 = __intrinsic__dict_get3(%1, %2, %3)
394: %0 = __intrinsic__tuple_make_(%0)
395: %0 = __intrinsic__v_call2(%1, %0)
396: %1 = "a"
397: %2 = __intrinsic__get_global0()
398: %3 = "__builtin__global_get"
399: %4 = None
400: %2 = __intrinsic__dict_get3(%2, %3, %4)
401: %1 = __intrinsic__tuple_make_(%1)
402: %1 = __intrinsic__v_call2(%2, %1)
403: %2 = __intrinsic__tuple_make_()
404: %3 = __intrinsic__get_global0()
405: %4 = "__builtin__call"
406: %5 = None
407: %3 = __intrinsic__dict_get3(%3, %4, %5)
408: %1 = __intrinsic__tuple_make_(%1, %2)
409: %1 = __intrinsic__v_call2(%3, %1)
410: %2 = "a"
411: %3 = __intrinsic__get_global0()
412: %4 = "__builtin__global_get"
413: %5 = None
414: %3 = __intrinsic__dict_get3(%3, %4, %5)
415: %2 = __intrinsic__tuple_make_(%2)
416: %2 = __intrinsic__v_call2(%3, %2)
417: %3 = __intrinsic__tuple_make_()
418: %4 = __intrinsic__get_global0()
419: %5 = "__builtin__call"
420: %6 = None
421: %4 = __intrinsic__dict_get3(%4, %5, %6)
422: %2 = __intrinsic__tuple_make_(%2, %3)
423: %2 = __intrinsic__v_call2(%4, %2)
424: %1 = __intrinsic__tuple_make_(%1, %2)
425: %2 = __intrinsic__get_global0()
426: %3 = "__builtin__call"
427: %4 = None
428: %2 = __intrinsic__dict_get3(%2, %3, %4)
429: %0 = __intrinsic__tuple_make_(%0, %1)
430: %0 = __intrinsic__v_call2(%2, %0)
431: %0 = "print"
432: %1 = __intrinsic__get_global0()
433: %2 = "__builtin__global_get"
434: %3 = None
435: %1 = __intrinsic__dict_get3(%1, %2, %3)
436: %0 = __intrinsic__tuple_make_(%0)
437: %0 = __intrinsic__v_call2(%1, %0)
438: %1 = "b"
439: %2 = __intrinsic__get_global0()
440: %3 = "__builtin__global_get"
441: %4 = None
442: %2 = __intrinsic__dict_get3(%2, %3, %4)
443: %1 = __intrinsic__tuple_make_(%1)
444: %1 = __intrinsic__v_call2(%2, %1)
445: %2 = __intrinsic__tuple_make_()
446: %3 = __intrinsic__get_global0()
447: %4 = "__builtin__call"
448: %5 = None
449: %3 = __intrinsic__dict_get3(%3, %4, %5)
450: %1 = __intrinsic__tuple_make_(%1, %2)
451: %1 = __intrinsic__v_call2(%3, %1)
452: %1 = __intrinsic__tuple_make_(%1)
453: %2 = __intrinsic__get_global0()
454: %3 = "__builtin__call"
455: %4 = None
456: %2 = __intrinsic__dict_get3(%2, %3, %4)
457: %0 = __intrinsic__tuple_make_(%0, %1)
458: %0 = __intrinsic__v_call2(%2, %0)
459: %0 = "print"
460: %1 = __intrinsic__get_global0()
461: %2 = "__builtin__global_get"
462: %3 = None
463: %1 = __intrinsic__dict_get3(%1, %2, %3)
464: %0 = __intrinsic__tuple_make_(%0)
465: %0 = __intrinsic__v_call2(%1, %0)
466: %1 = "a"
467: %2 = __intrinsic__get_global0()
468: %3 = "__builtin__global_get"
469: %4 = None
470: %2 = __intrinsic__dict_get3(%2, %3, %4)
471: %1 = __intrinsic__tuple_make_(%1)
472: %1 = __intrinsic__v_call2(%2, %1)
473: %2 = __intrinsic__tuple_make_()
474: %3 = __intrinsic__get_global0()
475: %4 = "__builtin__call"
476: %5 = None
477: %3 = __intrinsic__dict_get3(%3, %4, %5)
478: %1 = __intrinsic__tuple_make_(%1, %2)
479: %1 = __intrinsic__v_call2(%3, %1)
480: %1 = __intrinsic__tuple_make_(%1)
481: %2 = __intrinsic__get_global0()
482: %3 = "__builtin__call"
483: %4 = None
484: %2 = __intrinsic__dict_get3(%2, %3, %4)
485: %0 = __intrinsic__tuple_make_(%0, %1)
486: %0 = __intrinsic__v_call2(%2, %0)


Closure #1(Local size = 5):
0: %2 = __intrinsic__v_args0()
1: %3 = __intrinsic__tuple_len1(%2)
2: %4 = 0
3: %4 = __intrinsic__tuple_idx2(%2, %4)
4: %0 = %4
5: %2 = __intrinsic__get_global0()
6: %1 = %2
7: %2 = None
8: %2 = __intrinsic__dict_get3(%1, %0, %2)
9: Return %2
10: %2 = None
11: Return %2


Closure #2(Local size = 5):
0: %2 = __intrinsic__v_args0()
1: %3 = __intrinsic__tuple_len1(%2)
2: %4 = 0
3: %4 = __intrinsic__tuple_idx2(%2, %4)
4: %0 = %4
5: %4 = 1
6: %4 = __intrinsic__tuple_idx2(%2, %4)
7: %1 = %4
8: %2 = __intrinsic__v_call2(%0, %1)
9: Return %2
10: %2 = None
11: Return %2


Closure #3(Local size = 5):
0: %2 = __intrinsic__v_args0()
1: %3 = __intrinsic__tuple_len1(%2)
2: %4 = 0
3: %4 = __intrinsic__tuple_idx2(%2, %4)
4: %0 = %4
5: %4 = 1
6: %4 = __intrinsic__tuple_idx2(%2, %4)
7: %1 = %4
8: %2 = __intrinsic__str_concat2(%0, %1)
9: Return %2
10: %2 = None
11: Return %2


Closure #4(Local size = 5):
0: %2 = __intrinsic__v_args0()
1: %3 = __intrinsic__tuple_len1(%2)
2: %4 = 0
3: %4 = __intrinsic__tuple_idx2(%2, %4)
4: %0 = %4
5: %4 = 1
6: %4 = __intrinsic__tuple_idx2(%2, %4)
7: %1 = %4
8: %2 = __intrinsic__tuple_idx2(%0, %1)
9: Return %2
10: %2 = None
11: Return %2


Closure #5(Local size = 8):
0: %1 = __intrinsic__v_args0()
1: %2 = __intrinsic__tuple_len1(%1)
2: %3 = 0
3: %3 = __intrinsic__int_lt2(%3, %2)
4: JmpIfFalse %3 <10>
5: %5 = 0
6: %5 = __intrinsic__tuple_idx2(%1, %5)
7: %4 = %5
8: %5 = False
9: JmpIfFalse %5 <14>
10: %6 = __intrinsic__v_defaults0()
11: %7 = 0
12: %6 = __intrinsic__tuple_idx2(%6, %7)
13: %4 = %6
14: %0 = %4
15: %1 = "\n"
16: %2 = __intrinsic__get_global0()
17: %3 = "__builtin__add"
18: %4 = None
19: %2 = __intrinsic__dict_get3(%2, %3, %4)
20: %1 = __intrinsic__tuple_make_(%0, %1)
21: %1 = __intrinsic__v_call2(%2, %1)
22: %0 = %1
23: %1 = __intrinsic__print1(%0)
24: %1 = 233
25: Return %1
26: %1 = None
27: Return %1


Closure #6(Local size = 7):
0: %1 = __intrinsic__v_args0()
1: %0 = %1
2: %1 = "print_s"
3: %2 = __intrinsic__get_global0()
4: %3 = "__builtin__global_get"
5: %4 = None
6: %2 = __intrinsic__dict_get3(%2, %3, %4)
7: %1 = __intrinsic__tuple_make_(%1)
8: %1 = __intrinsic__v_call2(%2, %1)
9: %2 = "repr"
10: %3 = __intrinsic__get_global0()
11: %4 = "__builtin__global_get"
12: %5 = None
13: %3 = __intrinsic__dict_get3(%3, %4, %5)
14: %2 = __intrinsic__tuple_make_(%2)
15: %2 = __intrinsic__v_call2(%3, %2)
16: %3 = __intrinsic__tuple_make_(%0)
17: %4 = __intrinsic__get_global0()
18: %5 = "__builtin__call"
19: %6 = None
20: %4 = __intrinsic__dict_get3(%4, %5, %6)
21: %2 = __intrinsic__tuple_make_(%2, %3)
22: %2 = __intrinsic__v_call2(%4, %2)
23: %2 = __intrinsic__tuple_make_(%2)
24: %3 = __intrinsic__get_global0()
25: %4 = "__builtin__call"
26: %5 = None
27: %3 = __intrinsic__dict_get3(%3, %4, %5)
28: %1 = __intrinsic__tuple_make_(%1, %2)
29: %1 = __intrinsic__v_call2(%3, %1)
30: %1 = None
31: Return %1


Closure #7(Local size = 4):
0: %1 = __intrinsic__v_args0()
1: %2 = __intrinsic__tuple_len1(%1)
2: %3 = 0
3: %3 = __intrinsic__tuple_idx2(%1, %3)
4: %0 = %3
5: %1 = __intrinsic__repr1(%0)
6: Return %1
7: %1 = None
8: Return %1


Closure #8(Local size = 2):
0: %1 = __intrinsic__tuple_make_()
1: %1 = __intrinsic__tuple_make_(%1)
2: %1 = Closure #9() defaults = %1
3: %0 = %1
4: Return %0
5: %1 = None
6: Return %1


Closure #9(Local size = 8):
0: %1 = __intrinsic__v_args0()
1: %2 = __intrinsic__tuple_len1(%1)
2: %3 = 0
3: %3 = __intrinsic__int_lt2(%3, %2)
4: JmpIfFalse %3 <10>
5: %5 = 0
6: %5 = __intrinsic__tuple_idx2(%1, %5)
7: %4 = %5
8: %5 = False
9: JmpIfFalse %5 <14>
10: %6 = __intrinsic__v_defaults0()
11: %7 = 0
12: %6 = __intrinsic__tuple_idx2(%6, %7)
13: %4 = %6
14: %0 = %4
15: %1 = 0
16: %2 = 0
17: %3 = 1
18: %3 = __intrinsic__tuple_make_(%3)
19: %1 = __intrinsic__tuple_splice4(%0, %1, %2, %3)
20: Return %0
21: %1 = None
22: Return %1


Closure #10(Local size = 9):
0: %2 = __intrinsic__v_args0()
1: %3 = __intrinsic__tuple_len1(%2)
2: %4 = 0
3: %4 = __intrinsic__int_lt2(%4, %3)
4: JmpIfFalse %4 <10>
5: %6 = 0
6: %6 = __intrinsic__tuple_idx2(%2, %6)
7: %5 = %6
8: %6 = False
9: JmpIfFalse %6 <14>
10: %7 = __intrinsic__v_defaults0()
11: %8 = 0
12: %7 = __intrinsic__tuple_idx2(%7, %8)
13: %5 = %7
14: %0 = %5
15: %4 = 1
16: %5 = None
17: %6 = None
18: %2 = __intrinsic__tuple_slice4(%2, %4, %5, %6)
19: %1 = %2
20: %2 = "print"
21: %3 = __intrinsic__get_global0()
22: %4 = "__builtin__global_get"
23: %5 = None
24: %3 = __intrinsic__dict_get3(%3, %4, %5)
25: %2 = __intrinsic__tuple_make_(%2)
26: %2 = __intrinsic__v_call2(%3, %2)
27: %3 = __intrinsic__tuple_make_(%0, %1)
28: %4 = __intrinsic__get_global0()
29: %5 = "__builtin__call"
30: %6 = None
31: %4 = __intrinsic__dict_get3(%4, %5, %6)
32: %2 = __intrinsic__tuple_make_(%2, %3)
33: %2 = __intrinsic__v_call2(%4, %2)
34: %2 = None
35: Return %2


Hello, world!
ret = 233

(1, 1)
(1, (1, 1))
((1, 1), 1)

("a", ())
("Hello", ())
("Hello", ("world",))
("Hello", ("cruel", "world"))

((1, 1), (1, 1))
((1,),)
((1, 1, 1),)
Done
